### ISSUE TITLE
cpu load

### ISSUE THREAD

**Issue by joaoespanhol** [pterodactyl/panel#4623]
### Current Behavior

cpu load high use magma

### Expected Behavior

use same cpu load forge.

### Steps to Reproduce

is problem magma You can't check imagem when the server fully starts it uses less 100% forge, but magma uses above 100%

Magma
![image](https://user-images.githubusercontent.com/22871432/208265890-0c3cb840-5e86-4848-ac33-55a8dda96a63.png)

Forge
![image](https://user-images.githubusercontent.com/22871432/208265897-41c8791c-2701-4a75-af9a-a129960d3069.png)

when he says the following sentence: done for help type help.
problem forge %cpu drops to 30%.
Magma use 100%

observation team magma foundation:
[JustRed23](https://git.magmafoundation.org/JustRed23)
[@JustRed23](https://git.magmafoundation.org/JustRed23)
· [1 minute ago](https://git.magmafoundation.org/magmafoundation/Magma-1-18-x/-/issues/141#note_2759)
Maintainer
Spark usage shows only 30%. this might just be magma messing with pterodactyl server usage, but it only is 30%
more information time: https://git.magmafoundation.org/magmafoundation/Magma-1-18-x/-/issues/141

### Panel Version

1.11

### Wings Version

1.11

### Games and/or Eggs Affected

minecraft forge

### Docker Image

java 17

### Error Logs

```bash
more information: https://git.magmafoundation.org/magmafoundation/Magma-1-18-x/-/issues/141
```

### Is there an existing issue for this?

- [X] I have searched the existing issues before opening this issue.
- [X] I have provided all relevant details, including the specific game and Docker images I am using if this issue is related to running a server.
- [X] I have checked in the Discord server and believe this is a bug with the software, and not a configuration issue with my specific system.

**Comment by trenutoo** [comment]
Resource usage is for the container as a whole reported by Docker and displayed in the Panel. I don't see anything in this that's relevant to being a bug in the Pterodactyl software. 

Nor are different applications usually comparable resource usage wise, even if they're based on Forge. 100% is one thread.

**Comment by joaoespanhol** [comment]
> Resource usage is for the container as a whole reported by Docker and displayed in the Panel. I don't see anything in this that's relevant to being a bug in the Pterodactyl software.
> 
> Nor are different applications usually comparable resource usage wise, even if they're based on Forge. 100% is one thread.


I don't understand what you mean, can you explain it another way?

**Comment by joaoespanhol** [comment]
![image](https://user-images.githubusercontent.com/22871432/210441736-795769aa-d040-440b-8058-be1a7008aebb.png)

top inside the container vs outside the container on pterodactyl
THIS IS NOW CONFIRMED, and proven to not be an issue. Also note the total cpu usage in the bottom (system) top prompt being ~20%.

**Comment by joaoespanhol** [comment]
> Resource usage is for the container as a whole reported by Docker and displayed in the Panel. I don't see anything in this that's relevant to being a bug in the Pterodactyl software.
> 
> Nor are different applications usually comparable resource usage wise, even if they're based on Forge. 100% is one thread.

this is bug pterodacty
![image](https://user-images.githubusercontent.com/22871432/210441975-2b86f12f-07f4-4680-82ba-f79e990d0759.png)

top inside the container vs outside the container on pterodactyl
THIS IS NOW CONFIRMED, and proven to not be an issue. Also note the total cpu usage in the bottom (system) top prompt being ~20%.

**Comment by parkervcp** [comment]
>  THIS IS NOW CONFIRMED, and proven to not be an issue. Also note the total cpu usage in the bottom (system) top prompt being ~20%.

Considering the output here are formatted in vastly different ways but also an unclear amount of uptime on those due to that.

are you sure those are even the same processes? You may be seeing java from another container all together.

**Comment by joaoespanhol** [comment]
> > THIS IS NOW CONFIRMED, and proven to not be an issue. Also note the total cpu usage in the bottom (system) top prompt being ~20%.
> 
> Considering the output here are formatted in vastly different ways but also an unclear amount of uptime on those due to that.
> 
> are you sure those are even the same processes? You may be seeing java from another container all together.


yes, that's correct.
Magma also confirmed and other devs also confirmed this.

**Comment by joaoespanhol** [comment]
tested outside pterodactyl, do not use above 100% cpu, when using pterodacty use

![image](https://github.com/pterodactyl/panel/assets/22871432/32ce13cf-cf68-4cee-bab5-da83bcd0514b)

**Comment by parkervcp** [comment]
I am going to say that the values you are looking at are most likely a percentage out of the entire machine and not the specific application. This would mean that you are seeing 5% of the whole cpu and not a single thread that the 100% represents.

**Comment by C0D3-M4513R** [comment]
Magma dev here, and I can confirm that this issue exists. I have diagnosed it myself a while back too.

**Comment by C0D3-M4513R** [comment]
Outside docker
![Capture](https://github.com/pterodactyl/panel/assets/28912031/b31de710-fd3b-441f-8cf3-6d56b1777b1a)
![Capture1](https://github.com/pterodactyl/panel/assets/28912031/ce263af1-30b2-491d-8ff7-e75f6edf035d)
Inside Docker
![top](https://github.com/pterodactyl/panel/assets/28912031/e91b51f2-869f-4853-ac05-8a2d0810d3a0)


But If I were to go in with jmx then I'd see that it is not doing anything

**Comment by C0D3-M4513R** [comment]
Here is jmx/rmi and top in the container together: Java is not doing anything.
![Capture](https://github.com/pterodactyl/panel/assets/28912031/fbaecfb1-95d6-4eba-a3fa-e5fb7ea96fed)
Here I did a CPU profiling. Notice how all the threads that take active cpu time are sleeping?
Java readLine is blocking and will/should wait for a new line to be ready.
![Capture](https://github.com/pterodactyl/panel/assets/28912031/0171bdae-ae61-4f72-a041-a9cb03af32e9)
And here are the other threads
![Capture1](https://github.com/pterodactyl/panel/assets/28912031/40222376-5258-42c7-90da-210a98407df0)

**Comment by C0D3-M4513R** [comment]
Okay, since all those Numbers suggested, that magma used 100%, I investigated a little, and found something. Still have to see if the main Magma dev approves of my changes.

Tracked here: https://git.magmafoundation.org/magmafoundation/Magma-1-18-x/-/merge_requests/261

Before that Merge Request, Magma took 14% on my pc (Windows). After it only takes 0.3%.
So this issue can be closed.

**Comment by C0D3-M4513R** [comment]
This should now be solved by Magma 1.18 with the commit 7d684984
